### DISCUSSION TITLE
How can I use CPU only?

### DISCUSSION THREAD

**Discussion by Kadenza44** [xinntao/Real-ESRGAN#706] [Q&A]
I have a powerful server without GPU. Can I train a model on my data on the CPU?

I receive the message: 
RuntimeError: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx

**Comment by 10and01** [comment]
you could download the cpu version pytorch through https://pytorch.org/ (for me I use anaconda to deploy the cpu version) and choose the corresponding commands to do so, just as the image below shows.
![image](https://github.com/xinntao/Real-ESRGAN/assets/148670018/c9538a64-fe6c-49bb-9d27-8d799f2363b8)

**Comment by SA-j00u** [comment]
there is some hardcoded cuda() in code

**Reply by SA-j00u** [reply]
also i suspect that it doesn't utilize all CPU cores
(i'm using torch-2.0.1+cpu)
